app/lib/request.rb

```diff
@@ -199,7 +199,8 @@ def open(host, *args)
         rescue IPAddr::InvalidAddressError
           Resolv::DNS.open do |dns|
             dns.timeouts = 5
-            addresses = dns.getaddresses(host).take(2)
+            addresses = dns.getaddresses(host)
+            addresses = addresses.filter { |addr| addr.is_a?(Resolv::IPv6) }.take(2) + addresses.filter { |addr| !addr.is_a?(Resolv::IPv6) }.take(2)
```

Comment:
Hey guys! Isn't this two times IPv6? Shouldn't it be IPv6 + IPv4?

Reply:
There's a `!`!. But explicitly checking IPv4 instead of non-IPv4 would make sense too I guess.